flow/scripts/synth.tcl

```diff
@@ -5,8 +5,13 @@ if { [info exist ::env(SYNTH_HIERARCHICAL)] && $::env(SYNTH_HIERARCHICAL) == 1 &
   source $::env(SYNTH_STOP_MODULE_SCRIPT)
 }
 
+if { [info exist ::env(SYNTH_GUT)] && $::env(SYNTH_GUT) == 1 } {
+  hierarchy -check -top $::env(DESIGN_NAME)
+  delete $::env(DESIGN_NAME)/c:*
```

Comment:
What does c:* match?

Reply:
all cells, which quickly optimizes away everything else, including macros. Added comment.

I was going to delete wires too, but then I ran into a yosys snag and as it turns out, it's enough to delete the cells, optimization takes care of the rest...

https://github.com/YosysHQ/yosys/issues/4082